src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Commands/Export-AzureAsInstanceLog.cs

```diff
@@ -0,0 +1,189 @@
+                serverName = uriResult.PathAndQuery.Trim('/');
+                if (string.Compare(AsAzureClientSession.Instance.Profile.Context.Environment.Name, uriResult.DnsSafeHost, StringComparison.InvariantCultureIgnoreCase) != 0)
+                {
+                    AsAzureClientSession.Instance.SetCurrentContext(
+                        new AsAzureAccount(),
+                        AsAzureClientSession.Instance.Profile.CreateEnvironment(uriResult.DnsSafeHost));
+                }
+            }
+            else
+            {
+                var currentContext = AsAzureClientSession.Instance.Profile.Context;
+                if (currentContext != null
+                    && AsAzureClientSession.AsAzureRolloutEnvironmentMapping.ContainsKey(currentContext.Environment.Name))
+                {
+                    throw new PSInvalidOperationException(string.Format(Resources.InvalidServerName, serverName));
+                }
+            }
+
+            if (this.AsAzureHttpClient == null)
+            {
+                this.AsAzureHttpClient = new AsAzureHttpClient(() =>
+                {
+                    return new HttpClient();
+                });
+            }
+
+            if (this.TokenCacheItemProvider == null)
+            {
+                this.TokenCacheItemProvider = new TokenCacheItemProvider();
+            }
+        }
+
+        protected override void InitializeQosEvent()
+        {
+            // nothing to do here.
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Instance, Resources.ExportingLogFromAnalysisServicesServer))
+            {
+                var context = AsAzureClientSession.Instance.Profile.Context;
+                AsAzureClientSession.Instance.Login(context, null);
+                string accessToken = this.TokenCacheItemProvider.GetTokenFromTokenCache(
+                    AsAzureClientSession.TokenCache, context.Account.UniqueId);
+
+                Uri logfileBaseUri =
+                    new Uri(string.Format("{0}{1}{2}", Uri.UriSchemeHttps, Uri.SchemeDelimiter, context.Environment.Name));
+
+                UriBuilder resolvedUriBuilder = new UriBuilder(logfileBaseUri);
+                resolvedUriBuilder.Host = ClusterResolve(logfileBaseUri, accessToken, serverName);
+
+                var logfileEndpoint = string.Format(
+                        (string) context.Environment.Endpoints[AsAzureEnvironment.AsRolloutEndpoints.LogfileEndpointFormat],
+                        serverName);
+
+                this.AsAzureHttpClient.resetHttpClient();
+                using (HttpResponseMessage message = AsAzureHttpClient.CallGetAsync(
+                    resolvedUriBuilder.Uri,
+                    logfileEndpoint,
+                    accessToken).Result)
+                {
+                    message.EnsureSuccessStatusCode();
+                    File.WriteAllText(this.OutputPath, message.Content.ReadAsStringAsync().Result);
```

Comment:
Instead of using File.WriteAllText, this chould be using AzureSession.Instance.DataStore.WriteFile.  Also, per discussion, this should check for the file existence and ask for overwrite (overriding prompts wirh Force)  using Force || ShouldContinue.

Use AzureSession.Instance.DataStore.FileExists to check for existence